nautilus_trader/data/messages.pyx

```diff
@@ -107,6 +107,8 @@ cdef class SubscribeData(DataCommand):
     ----------
     data_type : type
         The data type for the subscription.
+    instrument_id : InstrumentId
```

Comment:
I'm a little on the fence about making `instrument_id` optional and adding it to the more general subscribe custom data message. Does it make downstream logic in the catalog easier?

Reply:
As discussed, an `instrument_id` is not applicable to all types of data, but this streamlines the current logic in the catalog - so OK at this point.